workspaces/mcp-chat-client/plugins/mcp-chat-client-backend/src/router.ts

```diff
@@ -0,0 +1,304 @@
+      npxCommand: serverConfig.getOptionalString('npxCommand'),
+      args: serverConfig.getOptionalStringArray('args'),
+      env,
+      url: serverConfig.getOptionalString('url'),
+      headers,
+      type: serverConfig.getOptionalString('type') as
+        | 'stdio'
+        | 'sse'
+        | 'streamable-http'
+        | undefined,
+    };
+  });
+}
+
+export async function createRouter({
+  mcpClientService,
+  config,
+}: {
+  httpAuth: HttpAuthService;
+  mcpClientService: MCPClientService;
+  config: RootConfigService;
+}): Promise<express.Router> {
+  const router = Router();
+  router.use(express.json());
+
+  // Configuration status endpoint
+  router.get('/config/status', async (_req, res) => {
+    try {
+      const providerStatus = mcpClientService.getProviderStatus();
+      const mcpServers =
+        config.getOptionalConfigArray('mcpChat.mcpServers') || [];
+      const serverList = mcpServers.map(serverConfig => ({
+        id: serverConfig.getOptionalString('id'),
+        name: serverConfig.getString('name'),
+        type: serverConfig.getOptionalString('type') || 'stdio',
+        hasUrl: serverConfig.has('url'),
+        hasNpxCommand: serverConfig.has('npxCommand'),
+        hasScriptPath: serverConfig.has('scriptPath'),
+      }));
+
+      res.json({
+        provider: providerStatus,
+        mcpServers: serverList,
+      });
+    } catch (error) {
+      res.status(500).json({
+        error: 'Failed to get configuration status',
+        details: error instanceof Error ? error.message : 'Unknown error',
+      });
+    }
+  });
+
+  // MCP Chat route
+  router.post('/chat', async (req, res) => {
+    try {
+      const { messages, enabledTools = [] } = req.body;
+
+      // Validate messages array
+      if (!messages || !Array.isArray(messages) || messages.length === 0) {
+        return res.status(400).json({ error: 'No query provided' });
+      }
+
+      const userQuery = messages[messages.length - 1]?.content;
+
+      if (!userQuery) {
+        return res.status(400).json({ error: 'No query provided' });
+      }
+
+      const serverConfigs = loadServerConfigs(config);
+      await mcpClientService.initMCP(serverConfigs);
+
+      const { reply, toolCalls, toolResponses } =
+        await mcpClientService.processQuery(messages, enabledTools);
+
+      if (toolCalls.length > 0) {
+        const toolsUsed = toolCalls.map(call => call.function.name);
+
+        return res.json({
+          role: 'assistant',
+          content: reply,
+          toolResponses,
+          toolsUsed,
+        });
+      }
+      return res.json({
+        role: 'assistant',
+        content: reply,
+        toolResponses: [],
+        toolsUsed: [],
+      });
+    } catch (err) {
+      console.error('[MCP Error]', err);
+      return res.status(500).json({ error: 'Something went wrong' });
+    }
+  });
+
+  // Test route with fixed prompt
+  router.get('/test/latest-news', async (_req, res) => {
```

Comment:
can all `/test/*` routes be removed?

Reply:
Done, have removed this endpoint and renamed other according to their functions